[DOCS] Support for nested functions in ES|QL STATS...BY

abdonpijpelink opened the pull request:
Adds documentation for the support for nested function in STATS...BY and aggregate functions (https://github.com/elastic/elasticsearch/pull/104387)

abdonpijpelink committed:
Document nested expressions for stats

abdonpijpelink committed:
Merge upstream/main

abdonpijpelink committed:
More docs

elasticsearchmachine commented:
Pinging @elastic/es-docs (Team:Docs)

elasticsearchmachine commented:
Pinging @elastic/es-analytical-engine (Team:Analytics)

alex-spies reviewed (approved):
LGTM, very nice, esp. the added examples!

Please consider all my comments as minor and implement any suggestions at your own discretion.

leemthompo commented on docs/reference/esql/functions/avg.asciidoc:
```diff
@@ -20,7 +20,7 @@ The average of a numeric expression.
 
 The result is always a `double` no matter the input type.
```
> We could mention that individual null values are skipped when computing the avg. (This applies to all our aggs, so that could be also a disclaimer at the top or so.)

General disclaimer could go [here](https://www.elastic.co/guide/en/elasticsearch/reference/master/esql-functions-operators.html#esql-agg-functions)?

leemthompo committed:
Apply suggestions from review

- count-distinct.asciidoc
  - Content restructured, moving the section about approximate counts to end of doc.

- count.asciidoc
  - Clarified that omitting the `expression` parameter in `COUNT` is equivalent to `COUNT(*)`, which counts the number of rows.

- percentile.asciidoc
  - Moved the note about `PERCENTILE` being approximate and non-deterministic to end of doc.

- stats.asciidoc
  - Clarified the `STATS` command
  -  Added a note indicating that individual `null` values are skipped during aggregation

leemthompo committed:
Comment out mentioning a buggy behavior

leemthompo committed:
Update sum with inline function example, update test file

leemthompo commented on docs/reference/esql/functions/sum.asciidoc:
```diff
@@ -27,9 +27,9 @@ include::{esql-specs}/stats.csv-spec[tag=sum]
-The expression can use inline functions. For example, to calculate the number of
-values for a multivalued column, use `MV_COUNT` to first count the number of
-values per row, and use the result with the `SUM` function:
+The expression can use inline functions. For example, to calculate the total 
+sum of the maximum values for each set of multivalued entries in a column,
+apply the `MV_MAX` function to each set and then sum the results:
```
@alex-spies Pretty sure this attempt at an explanation could be simplified :)

alex-spies replied:
Maybe just
`For example, to calculate the sum of each employee's maximum salary changes, apply ...`?
I think I'd mostly avoid using `set` - I _think_ we don't normally use that to describe the values of a (multi-valued) row.

leemthompo replied:
Sounds good— cuts verbosity too ;)

leemthompo commented on docs/reference/esql/functions/sum.asciidoc:
```diff
@@ -26,3 +26,16 @@ include::{esql-specs}/stats.csv-spec[tag=sum]
+
+The expression can use inline functions. For example, to calculate the number of
+values for a multivalued column, use `MV_COUNT` to first count the number of
+values per row, and use the result with the `SUM` function:
```
Tried update in https://github.com/elastic/elasticsearch/pull/104788/commits/ec2545751507ab85abfbc375173c9850a945a275

leemthompo committed:
Fix typo

alex-spies commented on docs/reference/esql/functions/count.asciidoc:
```diff
@@ -12,12 +12,14 @@ COUNT([expression])
 *Description*
 
 Returns the total number (count) of input values.
+See also <<esql-agg-count-distinct-approximate>>.
```
Argh, sorry - I think my comment from before was incorrect.

I think counts are only approximate for `COUNT_DISTINCT`, not for `COUNT`. Let's leave this out, after all - my bad.

leemthompo replied:
No worries

leemthompo committed:
Delete line

alex-spies reviewed (approved):
The changes LGTM, thanks @leemthompo !

leemthompo committed:
Simplify wording

leemthompo committed:
Merge branch 'main' into docs-esql-stats-by-nested-functions

leemthompo committed:
Fix conflict fix typo
